aya/src/sys/bpf.rs

```diff
@@ -477,10 +477,14 @@ pub(crate) fn bpf_prog_get_fd_by_id(prog_id: u32) -> Result<OwnedFd, SyscallErro
     })
 }
 
-fn bpf_obj_get_info_by_fd<T>(fd: BorrowedFd<'_>) -> Result<T, SyscallError> {
+fn bpf_obj_get_info_by_fd<T, F: FnOnce(&mut T)>(
+    fd: BorrowedFd<'_>,
+    init: F,
+) -> Result<T, SyscallError> {
     let mut attr = unsafe { mem::zeroed::<bpf_attr>() };
-    // info gets entirely populated by the kernel
-    let info = MaybeUninit::zeroed();
+    let mut info = unsafe { mem::zeroed() };
```

Comment:
Can this stay as `MaybeUninit::zeroed();` ?

Reply:
How's that better? We'll need to immediately `assume_init` so we can write to it.

Reply:
why not just call `as_mut_ptr()` on `info` in the init closure? Since we only need to assume_init in that one case

Reply:
We'd still need to assume_init *later*, before returning. What is the benefit of using `MaybeUninit` here?

Reply:
Or just use as_mut_ptr() here and ONLY do unsafe code in that one case

Reply:
This doesn't make sense - look at the diff. Before the change, we had unsafe code on the return path.

Reply:
Writing through `MaybeUninit::as_mut_ptr` is nasty:

> Gets a mutable pointer to the contained value. Reading from this pointer or turning it into a reference is undefined behavior unless the MaybeUninit<T> is initialized.

https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#method.as_mut_ptr

That means we can't turn it into a mutable reference because that's UB. Again, how is this better?

Reply:
Fair enough I guess there is no way to protect against the unsafe code here 👍 In the docs it says 
```
This has the same effect as [MaybeUninit::zeroed().assume_init()](https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#method.zeroed). It is useful for FFI sometimes, but should generally be avoided.
```
I guess this is the "usefult for FFI case" lol :)

Reply:
Yup.